### PR TITLE
Introduce new rule authconfig_config_files_symlinks

### PR DESCRIPTION
**Pull request by freddieRv**
#### Description:

- This new rule covers the DISA STIG requirement OL07-00-010199.
- Add OVAL checks and a few tests for this new rule.
- Also include this rule in OL7 and RHEL7 stig profiles.

#### Rationale:

- OL7/RHEL7 STIG efforts. This requirement was added to the OL7 DISA STIG in the v2r10 revision and RHEL7 DISA STIG v3r10 one.

#### Review Hints:

- Here is the fix text from DISA:
```
Rename the existing configuration files (skip this step if symbolic links are already present):
     $ sudo mv /etc/pam.d/system-auth /etc/pam.d/system-auth-ac
     $ sudo mv /etc/pam.d/password-auth /etc/pam.d/password-auth-ac
	
Create custom system-auth configuration file:
     $ sudo vi /etc/pam.d/system-auth-local
	
The new file, at minimum, must contain the following lines:
	
auth        required      pam_faillock.so preauth silent audit deny=3 even_deny_root fail_interval=900 unlock_time=900
auth        include       system-auth-ac
auth        sufficient    pam_unix.so try_first_pass
auth        [default=die] pam_faillock.so authfail audit deny=3 even_deny_root fail_interval=900 unlock_time=900

account     required      pam_faillock.so
account     include       system-auth-ac

password    requisite     pam_pwhistory.so use_authtok remember=5 retry=3
password    include       system-auth-ac
password    sufficient    pam_unix.so sha512 shadow try_first_pass use_authtok

session     include       system-auth-ac

Create custom password-auth configuration file:
     $ sudo vi /etc/pam.d/password-auth-local
	
The new file, at minimum, must contain the following lines:

auth        required      pam_faillock.so preauth silent audit deny=3 even_deny_root fail_interval=900 unlock_time=900
auth        include       password-auth-ac
auth        sufficient    pam_unix.so try_first_pass
auth        [default=die] pam_faillock.so authfail audit deny=3 even_deny_root fail_interval=900 unlock_time=900

account     required      pam_faillock.so
account     include       password-auth-ac

password    requisite     pam_pwhistory.so use_authtok remember=5 retry=3
password    include       password-auth-ac
password    sufficient    pam_unix.so sha512 shadow try_first_pass use_authtok

session     include       password-auth-ac
	
Create new or move existing symbolic links to the new custom configuration files:
     $ sudo ln -sf /etc/pam.d/system-auth-local /etc/pam.d/system-auth
     $ sudo ln -sf /etc/pam.d/password-auth-local /etc/pam.d/password-auth

Once finished, the file structure should be the following:
    $ sudo ls -1 /etc/pam.d/{password,system}-auth*

    /etc/pam.d/password-auth
    /etc/pam.d/password-auth-ac
    /etc/pam.d/password-auth-local
    /etc/pam.d/system-auth
    /etc/pam.d/system-auth-ac
    /etc/pam.d/system-auth-local
```
- While the introduced OVAL content does check what the DISA requirement states. I would like to discuss the feasibility of automated remediations. I think the tricky part is to make sure that the `-local` files properly include the `-ac` ones.
- Also I believe remediating this requirement would not impact the currently available remediations which edit PAM files

### CODE REVIEW

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/rule.yml
Comment (marcusburghardt) [inline linux_os/guide/system/auditing/authconfig_config_files_symlinks/rule.yml:5]: Could you ensure lines no longer than 99 columns, please?
https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#all-files

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/oval/shared.xml
Comment (marcusburghardt) [inline linux_os/guide/system/auditing/authconfig_config_files_symlinks/oval/shared.xml:26]: Wherever possible, could you ensure lines no longer than 99 columns, please?
https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#all-files

**Review by marcusburghardt — CHANGES_REQUESTED** [review]
Regarding the remediation, I believe it is too risky an automation with so many changes in PAM. PAM files are very sensible to ordering and custom PAM files could easily make it impossible to safely find the correct patterns.

My suggestion would be skip the remediation for this rule and include a warning in the `rule.yml` informing this remediation should be carefully and manually done by the administration, in alignment to the site policy.

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/rule.yml
Comment (freddieRv) [reply linux_os/guide/system/auditing/authconfig_config_files_symlinks/rule.yml]: Addressed

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/oval/shared.xml
Comment (freddieRv) [reply linux_os/guide/system/auditing/authconfig_config_files_symlinks/oval/shared.xml]: Addressed

File: shared/references/cce-redhat-avail.txt
Comment (freddieRv) [reply shared/references/cce-redhat-avail.txt:1]: Here is were they are used:
- CCE-86035-3, CCE-86036-1 -> `mount_option_home_usrquota`
- CCE-86037-9,  CCE-86039-5 -> `mount_option_home_grpquota`

I thought it was a mistake that they were still present here.

Comment (marcusburghardt) [reply shared/references/cce-redhat-avail.txt:1]: You are correct. The were properly removed by f3ae6683846 but somehow included again by 82bd9ad4329.
Thanks for catching this.

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/correct_set-up.pass.sh
Comment (marcusburghardt) [inline linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/correct_set-up.pass.sh:40]:
```suggestion
ln -sf /etc/pam.d/system-auth-local /etc/pam.d/system-auth
```

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/correct_set-up.pass.sh
Comment (marcusburghardt) [inline linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/correct_set-up.pass.sh:41]:
```suggestion
ln -sf /etc/pam.d/password-auth-local /etc/pam.d/password-auth
```

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/correct_set-up.pass.sh
Comment (marcusburghardt) [reply linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/correct_set-up.pass.sh:40]: The file might be already a symlink.

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/symlinks_wrong_target.fail.sh
Comment (marcusburghardt) [inline linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/symlinks_wrong_target.fail.sh:40]:
```suggestion
ln -sf /etc/pam.d/system-auth-mycustomconfig /etc/pam.d/system-auth
```

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/symlinks_wrong_target.fail.sh
Comment (marcusburghardt) [inline linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/symlinks_wrong_target.fail.sh:41]:
```suggestion
ln -sf /etc/pam.d/password-auth-mycustomconfig /etc/pam.d/password-auth
```

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/oval/shared.xml
Comment (marcusburghardt) [inline linux_os/guide/system/auditing/authconfig_config_files_symlinks/oval/shared.xml:3]: I feel the OVAL description should be more aligned to the message in the "rule title". What do you think?

**Review by marcusburghardt — CHANGES_REQUESTED** [review]

**Comment by marcusburghardt** [comment]
FYI @yuumasato

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/correct_set-up.pass.sh
Comment (freddieRv) [reply linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/correct_set-up.pass.sh]: addressed

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/correct_set-up.pass.sh
Comment (freddieRv) [reply linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/correct_set-up.pass.sh]: addressed

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/symlinks_wrong_target.fail.sh
Comment (freddieRv) [reply linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/symlinks_wrong_target.fail.sh]: addressed

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/symlinks_wrong_target.fail.sh
Comment (freddieRv) [reply linux_os/guide/system/auditing/authconfig_config_files_symlinks/tests/symlinks_wrong_target.fail.sh]: addressed

File: linux_os/guide/system/auditing/authconfig_config_files_symlinks/oval/shared.xml
Comment (freddieRv) [reply linux_os/guide/system/auditing/authconfig_config_files_symlinks/oval/shared.xml]: I used the rule title as the description here. I'm open to suggestions on a better wording.

Comment (marcusburghardt) [reply linux_os/guide/system/auditing/authconfig_config_files_symlinks/oval/shared.xml]: Thanks @freddieRv . It is better now IMO.

**Review by marcusburghardt — APPROVED** [review]
LGTM. Thanks @freddieRv

**Comment by marcusburghardt** [comment]
Automatus CS8 and Automatus CS9 are failing because the rule prodtype is limited to ol7 and rhel7.

**Comment by marcusburghardt** [comment]
I am overriding the CODEOWNERS file as @freddieRv can not merge this PR.

**Comment by vojtapolasek** [comment]
@freddieRv  Hello, would you please move this rule into a different group? I think it belongs more into "accounts" than "auditing".

**Pull request merged**
